pyam/compute.py

```diff
@@ -249,6 +251,248 @@ def bias(self, name, method, axis):
         """
         _compute_bias(self._df, name, method, axis)
 
+    def kaya_variables(self, scenarios, append=False):
+        """Compute the variables needed to compute Kaya factors
+        for the Kaya Decomposition Analysis.
+
+        Parameters
+        ----------
+        scenarios : iterable of tuples (model, scenario, region)
+            The (model, scenario, region) combinations to be included.
```

Comment:
I removed the scenarios filter and followed suggestion #1--we'll try to compute kaya-variables for all scenario/model/region combinations in the dataframe. I didn't try to implement returning partial sets of kaya-variables for scenario/model/region combinations that have only partial input data (though, if you think this is an important feature I'll give it a try). There's a new helper function in the kaya_variables module, _is_input_data_incomplete(), that relies on the require_data() method to make sure the data is complete enough so that none of the arithmetic operations in later processing will throw errors. 

require_data() only considers model/scenario combinations (it does not care if all variables are present in all regions), but I want to alert the user when regions without full input data are present, so there is extra logic in _is_input_data_incomplete() to log this information.